### ISSUE TITLE
IslandBanEvent: players can ban players who are not affiliated with their island

### ISSUE THREAD

**Issue by Evalutik** [BG-Software-LLC/SuperiorSkyblock2#1795]
### Minecraft's Version

PurPur 1.19.3

### Plugin's Version

SuperiorSkyblock2-2023.2-b45

### Describe the bug

If a player has an island, they can simply select any other player and spam them with /is ban & /is unban
I want to notify a player that they've been banned, but I don't want any player to be able to ban any player
![image](https://github.com/BG-Software-LLC/SuperiorSkyblock2/assets/126126362/72ac41d8-16f3-4c5d-846b-598c8f671f8e)

It would be much better to have the following:
`/is banvisitor <player>` - the command blocks and kicks from the island the player who is the **visitor of the island at the moment**. **The  command cannot block anyone other than visitors**. You will still be able to invite this player to your island as a teammate, however, this player will not be able to visit your island if he is not a member of it.
`/is unbanvisitor <player>` - the command allows blocked from visit player to be a visitor on your island.
`/is ban <player>` - the command blocks and kicks from the island the player who is the **teammate** of your island. **It can't affect on any other players.** The player cannot be invited to the island until it is unlocked. The blocked player cannot visit the island after that. To unlock access to visit the island, you must use the command /is unbanvisitor <player>
`/is unban <player>` - If the player cannot visit the island, or cannot be invited to the island, or cannot visit the island AND cannot be invited to the island, then after using this command, this can be done.

Events:
`IslandBanEvent` - only after `/is ban` command
`IslandBanVisitotEvent` - after `/is banvisitor` and `/is ban commands`
`IslandUnbanEvent` - only after `/is unban` command
`IslandUnbanVisitorEvent` - after `/is unbanvisitor` and `/is unban commands`

### To Reproduce

you can just ban player which isnot related to you island

**Comment by Evalutik** [comment]
In general, I want to make sure that the player's inventory is cleared every time they leave the island, and the same way every time they enter the island. So that the player always joins "clean" and is also "clean" in the hub.
So I added the these lines to event-commands in config.yml:
```
  IslandCreateEvent:
  - island admin cmdall %player% false invclear {player-name}
  IslandDisbandEvent:
  - island admin cmdall %player% false invclear {player-name}
  IslandJoinEvent:
  - invclear %player%
  IslandQuitEvent:
  - invclear %player%
  IslandKickEvent:
  - invclear %target%
  **IslandBanEvent:
  - invclear %target%**
```
But now I've discovered that players can trigger the IslandBanEvent for any player at any time. So I need to remove these lines:
```
  **IslandBanEvent:
  - invclear %target%**
```
_In the same way as the examples above, I clear experience, ender chest, effects, etc..._
But how do I clear a player's inventory if they were on an island team and then banned? Players must not clear the inventories of island visitors and other random players at will with the /is ban command

Have I taken into account all the events, or can a player join / leave from the island in other ways? _(Note: in my configuration, visitors of the island cannot pick up/interact with items in any way, so they do NOT need to be cleared)_ 
thx!

**Comment by Evalutik** [comment]
In addition, I noticed that a similar situation with coops. Players can add coops without confirming who they are adding. Please add a co-op request instead so that the player can accept or reject the invitation to be a co-op.

**Comment by OmerBenGera** [comment]
It makes no sense to not be able to ban players that are not part of the island.
Banning players makes them unable to join the island, visit it, etc.

If players abuse the notifications, you can either add a cooldown to the command or remove the notification.

Not a bug, closing this.

**Comment by OmerBenGera** [comment]
> In addition, I noticed that a similar situation with coops. Players can add coops without confirming who they are adding. Please add a co-op request instead so that the player can accept or reject the invitation to be a co-op. And I also do not understand how a player can independently leave the coop of another island.

Same here, why would people need to accept the coop?
